firebase-firestore/src/main/java/com/google/firebase/firestore/model/MutableDocument.java

```diff
@@ -17,6 +17,15 @@
 import androidx.annotation.NonNull;
 import com.google.firestore.v1.Value;
 
+/**
+ * Represents a document in Firestore with a key, version, data and whether it has local mutations
+ * applied to it. Documents can start out as {@link DocumentType#INVALID} and transition to a valid
```

Comment:
"Documents can start out as..." is not as relevant now since factory methods create the object directly into the desired state. Also, `DocumentType` is a private implementation detail that should not be mentioned in the javadocs for the class. This entire paragraph about starting in an invalid state and transitioning to a different state can probably just be deleted.

Reply:
Updated the comment.